### PR TITLE
middleware/ taskApp - taskAPI

### PR DESCRIPTION
**Pull request by VGamezz19**
TaskApp with middleware, for practice! 

````sh
├── package.json
├── src
│   ├── api
│   │   └── apiTask.js
│   ├── controller
│   │   └── taskApp.js
│   └── index.js
├── views
│   └── index.pug
└── yarn.lock
````

### CODE REVIEW

File: staff/vic/middleware-task-api/src/api/apiTask.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return axios.get(path)
+        .then(res => res.data)
+}
+
+function __post (path, text){
+    return axios.post(path, {text})
+}
+
+function __put (path){
+    return axios.put(path)
```
Comment (manuelbarzi) [inline staff/vic/middleware-task-api/src/api/apiTask.js]: MAY you return after .then(res => res.data), as axios encapsulates the response body there

File: staff/vic/middleware-task-api/src/api/apiTask.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+function __delete (path) {
+    return axios.delete(path)
```
Comment (manuelbarzi) [inline staff/vic/middleware-task-api/src/api/apiTask.js]: MAY you return after .then(res => res.data), as axios encapsulates the response body there

File: staff/vic/middleware-task-api/src/api/apiTask.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const URL = `http://localhost:5000/api`
+
+function __get (path) {
+    return axios.get(path)
+        .then(res => res.data)
+}
+
+function __post (path, text){
+    return axios.post(path, {text})
```
Comment (manuelbarzi) [inline staff/vic/middleware-task-api/src/api/apiTask.js]: MAY you return after .then(res => res.data), as axios encapsulates the response body there

File: staff/vic/middleware-task-api/src/index.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+const {redirect, getAllTasks, createTask, markDoneTask, deleteTask} = require('./controller/taskApp')
+
+const app = express()
+
+const formBodyParser = bodyParser.urlencoded({extended: false})
+
+app.set('view engine', 'pug')
+
+app.get('/', [getAllTasks])
```
Comment (manuelbarzi) [inline staff/vic/middleware-task-api/src/index.js]: MAY you group all common routes into a router?

**Comment by VGamezz19** [comment]
@manuelbarzi In this version, I have put all routes from `taskApp` in `/routes/task App Routes.js` and add some docs in `apiTasks.js`.

**Pull request merged**